src/argilla/server/contexts/datasets.py

```diff
@@ -494,9 +494,14 @@ def validate_record_fields(dataset: Dataset, fields: Dict[str, Any]):
             raise ValueError(f"Missing required value for field: {field.name!r}")
 
         value = fields_copy.pop(field.name, None)
-        if not isinstance(value, str):
+        if field.required and not isinstance(value, str):
```

Comment:
Since the behind condition already checks the required values, we can check type only when value is informed, so bug for optional fields will be fixed
```suggestion
        if value and not isinstance(value, str):
```